packages/EasyDecision/src/Bridge/Symfony/Messenger/ResetDecisionMiddleware.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyDecision\Bridge\Symfony\Messenger;
+
+use EonX\EasyDecision\Interfaces\DecisionFactoryInterface;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Middleware\MiddlewareInterface;
+use Symfony\Component\Messenger\Middleware\StackInterface;
+use Symfony\Component\Messenger\Stamp\ConsumedByWorkerStamp;
+
+final class ResetDecisionMiddleware implements MiddlewareInterface
+{
+    /**
+     * @var \EonX\EasyDecision\Interfaces\DecisionFactoryInterface
+     */
+    private $decisionFactory;
+
+    public function __construct(DecisionFactoryInterface $decisionFactory)
+    {
+        $this->decisionFactory = $decisionFactory;
+    }
+
+    public function handle(Envelope $envelope, StackInterface $stack): Envelope
+    {
+        if ($this->shouldSkip($envelope)) {
+            return $stack->next()
+                ->handle($envelope, $stack);
+        }
+
+        $this->decisionFactory->reset();
+
+        return $envelope;
```

Comment:
Here you're bypassing all the other middleware, we can't do that 😄 in this case we MUST always return the envelope processed by the rest of middleware.

So the logic should be:

- if message consumed by worker, then reset decision factory
- return stack next handle